services/endpointsrv/service.go

```diff
@@ -0,0 +1,454 @@
+//nolint:noctx,godox
+package endpointsrv
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/unweave/unweave/api/types"
+	"github.com/unweave/unweave/db"
+	"github.com/unweave/unweave/services/evalsrv"
+	"github.com/unweave/unweave/services/execsrv"
+	"go.jetpack.io/typeid"
+)
+
+type Driver interface {
+	EndpointDriverName() string
+	EndpointProvider() types.Provider
+	EndpointCreate(ctx context.Context, project, endpointID, execID string, internalPort int32) (string, error)
+}
+
+type Service interface {
+	EndpointExecCreate(ctx context.Context, projectID, execID string) (types.Endpoint, error)
+	EndpointAttachEval(ctx context.Context, endpointID, evalID string) error
+	EndpointList(ctx context.Context, projectID string) ([]types.Endpoint, error)
+	EndpointGet(ctx context.Context, id string) (types.Endpoint, error)
+	RunEndpointEvals(ctx context.Context, endpointID string) (string, error)
+	EndpointCheckStatus(ctx context.Context, checkID string) (types.EndpointCheck, error)
+}
+
+type EndpointService struct {
+	store  Store
+	evals  evalsrv.Service
+	execs  execsrv.Service
+	driver Driver
+}
+
+type Store interface {
+	EndpointCreate(ctx context.Context, arg db.EndpointCreateParams) error
+	EndpointGet(ctx context.Context, id string) (db.UnweaveEndpoint, error)
+	EndpointsForProject(ctx context.Context, id string) ([]db.UnweaveEndpoint, error)
+	EndpointEval(ctx context.Context, endpointID string) ([]db.UnweaveEndpointEval, error)
+	EndpointEvalAttach(ctx context.Context, arg db.EndpointEvalAttachParams) error
+	EndpointCheckCreate(ctx context.Context, arg db.EndpointCheckCreateParams) error
+	EndpointCheckStepCreate(ctx context.Context, arg db.EndpointCheckStepCreateParams) error
+	EndpointCheckStepUpdate(ctx context.Context, arg db.EndpointCheckStepUpdateParams) error
+	EndpointCheckSteps(ctx context.Context, checkID string) ([]db.UnweaveEndpointCheckStep, error)
+	EndpointCheck(ctx context.Context, checkID string) (db.UnweaveEndpointCheck, error)
```

Comment:
The original idea with the store was to wrap the db types and not leak them. This is good for now but we might want to update in the future